### ISSUE TITLE
Numbers in the series name recognized as chapter number if no chapter number in comicinfo.xml

### ISSUE THREAD

**Issue by CtrlAltDefeat94** [Kareadita/Kavita#2908]
### What happened?

When the series name  in the title contains a number such as `Fairy Tail: 100 Years Quest Vol. 1: New Start, New Quest`, this number is parsed as a chapter number. All volumes in this series are added for both their Volume number and Chapter 100.

### What did you expect?

If a cbz file has a comicinfo.xml and does not contain a Chapter tag, it should not add a chapter number.

### Kavita Version Number - If you don not see your version number listed, please update Kavita and see if your issue still persists.

0.8.1 - Stable

### What operating system is Kavita being hosted from?

Docker (Dockerhub Container)

### If the issue is being seen on Desktop, what OS are you running where you see the issue?

None

### If the issue is being seen on Mobile, what OS are you running where you see the issue?

None

**Comment by CtrlAltDefeat94** [comment]
Is there anything I can help with to get this out of the triage stage?
I'm currently unable to get (and keep) the Re Zero manga's in Kavita, as it gets confused by the Chapter in the series title.

**Comment by majora2007** [comment]
I need community help with this because partial ComicInfo is a feature many people need. You shouldn't have to tag everything perfectly to correct one issue. 

The problem comes from the filename parser, but it's not something that can be fixed per-se. My idea would be to use a unique symbol that would never be real that Kavita can read and ignore said field.

**Comment by majora2007** [comment]
@CtrlAltDefeat94 can you try for now as a temp workaround to tag the Number field as `-100000` and see if that works for you? It's not ideal, but it is the internal encoding that Kavita uses when there is no chapter number.

**Comment by Roshri** [comment]
I'm all in for a special value in comicinfo. So far Ive been tagging these as chapter 0 as a legacy measure.
I'm not sure about parser logic but shouldn't it discard a number if found before volume and/or chapter markers in the filename? AFAIK it should work for Manga naming standards

**Comment by majora2007** [comment]
Here is the situation with filename parsing:
`ABC Chapter 4 - A title.cbz`
and 
`"DEF Chapter 4 - A title.cbz`

These are both real titles. The first is the Re:ZERO Chapter 4 case, where Chapter 4 is part of the series name. Whereas `Bleach Chapter 4 - Death Gods` also is a valid case, where `Bleach` is the series name. 

From a filename parsing standpoint, there is no way to know the difference so there is no way to write a rule to accurately identify. The latter is the most common.

So Using a unique key, like `kavita` or `ignore` might be needed. Especially since the standard is pretty frozen.

**Comment by SleepingPanda** [comment]
Is it not possible to let the user manually override the parser?

**Comment by majora2007** [comment]
@SleepingPanda That is what tagging your files with metadata is doing. It's telling Kavita exactly what to take it as.

**Comment by SleepingPanda** [comment]
Right but tagging potentially hundreds of files may not always be a reasonable option. I'm asking if there is no way to allow that from within Kavita itself.

**Comment by majora2007** [comment]
I don't think so. I did a proof of concept to allow a library where the user defines the Regex, but it wouldn't necessarily solve this problem and was quite complex to support (and likely will be underused due to the complexities of parsing--the reason we have this issue to begin with). 

Just allowing loose filenames is not something I want the software to support nor something I want to be restricted to design around for future UX work.

**Comment by SleepingPanda** [comment]
There's a surprisingly wide variety of series with numbers in their names and I don't think tagging our files with a Kavita-specific symbol should be the long-term solution. There are use cases like mine (and a friend's) where files can't be modified for example.

**Comment by CtrlAltDefeat94** [comment]
I would find a setting useful to disable filename parsing for any file that contains a comicinfo.xml.
If the tag isnt in the comicinfo, it means that tag shouldnt be used at all.
For example Re Zero volumes with Chapter X in their titles dont have a chapter tag inside the comicinfo, so there's no need to parse it as such.

**Comment by majora2007** [comment]
> I would find a setting useful to disable filename parsing for any file that contains a comicinfo.xml. If the tag isnt in the comicinfo, it means that tag shouldnt be used at all. For example Re Zero volumes with Chapter X in their titles dont have a chapter tag inside the comicinfo, so there's no need to parse it as such.

I would be open to trying something like this. If you could open a FR and link back to this thread, that would help ensure it doesn't get lost.

**Comment by killo3967** [comment]
I think it might be time to use a local AI. I made a Python script that Ollama uses locally. First, I had to train it a bit. I gave it 200 examples of original comic book names, and how I wanted them, as well as the rules, and now it never fails. Perhaps Kavita should start considering the possible use of local AI for those users who want to use them. The truth is, it can be done very easily with a normal 8GB graphics card.

**Comment by daydreamrabbit** [comment]
I don't know why you approach the problem so hard. 

In fact, there are two very simple ways. 
1.Make the file recognized as a special piece as a short piece. 
2.If there is comicinfo.xml, give priority to the file name. 

It seems virtually impossible to use ai in reality.
